packages/offix-datastore/src/storage/IndexedDBStorage.ts

```diff
@@ -0,0 +1,102 @@
+import { Model, PersistedModel } from "../models";
+import { Storage } from "./Storage";
+import { getStoreNameFromModelName, generateId } from "./core";
+import { PredicateFunction } from "../predicates";
+
+const DB_NAME = "offix-datastore";
+
+export class IndexedDBStorage implements Storage {
+    public readonly indexedDB: Promise<IDBDatabase>;
+
+    constructor(models: Model[], schemaVersion: number) {
+        this.indexedDB = new Promise((resolve, reject) => {
+            const openreq = indexedDB.open(DB_NAME, schemaVersion);
+            openreq.onerror = () => reject(openreq.error);
+            openreq.onsuccess = () => {
+                const db = openreq.result;
+                db.onversionchange = function() {
+                    this.close();
+                    // alert("Please reload the page.");
+                };
+                resolve(db);
+            };
+
+            openreq.onupgradeneeded = () => {
```

Comment:
Speechless! You have even done support for the store upgrades! 
WOW. Just wow. I cannot wait to test this!

Reply:
Thank you! it just occurred to me. I comment out the alert because it causes errors during tests. There should be a way to resolve this though, I'm just putting off for now